Services/Administration/OJS.Services.Administration.Business/Roles/RolesBusinessService.cs

```diff
@@ -88,15 +99,33 @@ public async Task<string> GetIdByName(string roleName)
 
     private async Task<(string, UserProfile)> GetUserAndRoleName(UserToRoleModel model)
     {
-         var roleName = await this.roleDataService!
-            .GetByIdQuery(model.RoleId!)
-            .Select(x => x.Name!)
-            .FirstAsync();
+        var roleName = await this.roleDataService!
+           .GetByIdQuery(model.RoleId!)
+           .Select(x => x.Name!)
+           .FirstAsync();
+
+        var user = await this.usersDataService
+           .GetByIdQuery(model.UserId!)
+           .FirstAsync();
+
+        return (roleName, user);
+    }
+
+    private async Task RemoveUserFromLecturerInContestAndContestCategory(string roleName, UserProfile user)
+    {
+        if (roleName == Lecturer)
+        {
+            var categoriesUserIsLecturerIn = await this.lecturerInContestCategoryDataService
+                .GetQuery(licc => licc.LecturerId == user.Id)
+                .ToListAsync();
 
-         var user = await this.usersDataService
-            .GetByIdQuery(model.UserId!)
-            .FirstAsync();
+            this.lecturerInContestCategoryDataService.DeleteMany(categoriesUserIsLecturerIn);
 
-         return (roleName, user);
+            var contestsUserIsLecturerIn = await this.lecturerInContestsDataService
```

Comment:
This should be named userLecturerInContestsEntries or something like that..

Reply:
Done.